packages/@aws-cdk/aws-autoscaling/lib/scheduled-action.ts

```diff
@@ -75,9 +75,27 @@ export interface ScheduledActionProps extends BasicScheduledActionProps {
   readonly autoScalingGroup: IAutoScalingGroup;
 }
 
-const CRON_PART = '(\\*|\\?|[0-9]+)';
-
-const CRON_EXPRESSION = new RegExp('^' + [CRON_PART, CRON_PART, CRON_PART, CRON_PART, CRON_PART].join('\\s+') + '$');
+const CRON_MINUTES = `(\\*|([0-9]|[1-4][0-9]|5[0-9])|((([0-9]|[1-4][0-9]|5[0-9]))
+(\\,|\\/|\\-)(([0-9]|[1-4][0-9]|5[0-9]))))`;
+const CRON_HOURS = `(\\*|([0-9]|1[0-9]|2[0-3])|
+((([0-9]|1[0-9]|2[0-3]))(\\,|\\/|\\-)(([0-9]|1[0-9]|2[0-3]))))`;
+const CRON_DAY_OF_MONTH = `(\\*|\\?|L|([0-9]|1[0-9]|2[0-3])?W|
+([1-9]|[12][0-9]|3[01])|((([1-9]|[12][0-9]|3[01]))(\\,|\\/|\\-)(([1-9]|[12][0-9]|3[01]))))`;
+const CRON_MONTH = `(\\*|([1-9]|1[0-2]|JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)|
+(([1-9]|1[0-2]|JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)
+(\\,|\\/|\\-)([1-9]|1[0-2]|JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)))`;
+const CRON_DAY_OF_WEEK = `(\\*|\\?|L|[1-7]\\#[1-9]|
+([1-7]|MON|TUE|WED|THU|FRI|SAT|SUN)|(([1-7]|MON|TUE|WED|THU|FRI|SAT|SUN)
+(\\,|\\-)([1-7]|MON|TUE|WED|THU|FRI|SAT|SUN)))`;
+const CRON_YEAR = `(\\*|(19[78][0-9]|199[0-9]|20[0-9]{2}|21[0-8][0-9]|219[0-9])|
+(((19[78][0-9]|199[0-9]|20[0-9]{2}|21[0-8][0-9]|219[0-9]))(\\,|\\/|\\-)
+((19[78][0-9]|199[0-9]|20[0-9]{2}|21[0-8][0-9]|219[0-9]))))`;
+
+const CRON_EXPRESSION = new RegExp('^' +
+  [
+    CRON_MINUTES, CRON_HOURS, CRON_DAY_OF_MONTH, CRON_MONTH, CRON_DAY_OF_WEEK, CRON_YEAR
```

Comment:
Not sure where the YEAR field is coming from. This page says:

https://docs.aws.amazon.com/autoscaling/ec2/userguide/schedule_time.html#create-sch-actions-aws-cli

> You can specify a recurrence schedule, in UTC, using the Unix cron syntax format. This format consists of five fields separated by white spaces: [Minute] [Hour] [Day_of_Month] [Month_of_Year] [Day_of_Week]

This page also does not contain it: http://crontab.org/

Did you actually run the integ tests, or did you just update the expectations manually?

Reply:
I run them: I know the year is not present in cron, but I supposed from [here](https://docs.aws.amazon.com/autoscaling/application/userguide/application-auto-scaling-scheduled-scaling.html) that *extended cron* - the one used for rules - was supported. I think I looked at wrong doc page 😕If so, then we have to move to standard cron...

Reply:
Yeah. Application AutoScaling and AutoScaling are not the same.

Reply:
BTW, when you're changing it, can you explicitly add in there that although YEAR  is supported for appscaling, it's not supported for instance scaling?

Reply:
Ok, I will :) Thank you for your time!